core/src/main/java/bio/terra/pearl/core/service/portal/PortalEnvironmentService.java

```diff
@@ -137,11 +137,11 @@ public void delete(UUID id, Set<CascadeProperty> cascades) {
         portalParticipantUserService.deleteByPortalEnvironmentId(id);
         // clean up any preregistration responses not associated with a user
         preregistrationResponseDao.deleteByPortalEnvironmentId(id);
+        mailingListContactService.deleteByPortalEnvId(id);
```

Comment:
mailing list records can references participant user IDs, so the mailing list records need to be deleted first. we never really ran into this before because it's rare that a participant is signed in and then joins the mailing list